collector/lib/HostHeuristics.cpp

```diff
@@ -80,9 +80,39 @@ class DockerDesktopHeuristic : public Heuristic {
   }
 };
 
+class S390XHeuristic : public Heuristic {
+ public:
+  // S390X does not support eBPF ealier than 4.18.0-348 (rhel8.5) so we switch to use corebpf
+  // instead.
+  void Process(HostInfo& host, const CollectorConfig& config, HostConfig* hconfig) const {
+    auto k = host.GetKernelVersion();
+    std::string os_id = host.GetOSID();
+
+    if (k.machine != "s390x") {
+      return;
+    }
+
+    if (os_id == "rhel" || os_id == "centos") {
+      // example release version: 4.18.0-305.88.1.el8_4.s390x
+      // build_id = 305
+      if (k.release.find(".el8.") != std::string::npos) {
```

Comment:
This would not match the example release version given - does this mean that the example is wrong or should this line be something like `k.release.find(".el8")`?